static/js/components/dashboard/courses/util.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+import moment from 'moment';
+import R from 'ramda';
+
+import type { CourseRun } from '../../../flow/programTypes';
+import {
+  STATUS_OFFERED,
+  STATUS_PAID_BUT_NOT_ENROLLED,
+  STATUS_PASSED,
+  STATUS_MISSED_DEADLINE,
+  STATUS_NOT_PASSED,
+} from '../../../constants';
+import { S } from '../../../lib/sanctuary';
+
+export const courseStartDateMessage = (courseRun: CourseRun) => {
+  if (R.isNil(courseRun.course_start_date)) {
+    return "";
+  }
+
+  let startDate = moment(courseRun.course_start_date);
+  let daysUntilStart = startDate.diff(moment(), 'days');
+  if (daysUntilStart === 0) {
+    return "Course starts today";
+  } else if (daysUntilStart < 0) {
+    return `Course started ${startDate.format('M/D/YYYY')}`;
+  } else if (daysUntilStart < 10) {
+    return `Course starts in ${daysUntilStart} days`;
```

Comment:
FYI there's some support in moment for descriptive output though it's out of scope for this PR: https://github.com/moment/moment/issues/1035